benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -295,43 +304,145 @@ def update_ci_status(self, status: str, ci_minutes_time: int, benchmark_operator
+        :param database: database name
+        :param pod: False in case of vm
+        :param labels: list of labels of pod names - using it when it different from workload name
+        :return: run artifacts url
+        """
+        self.__create_pod_log(label='benchmark-controller-manager')
+        self.__create_pod_log(label='system-metrics')
+        # for vm call to create_vm_log
+        if pod:
+            # workload that contains 2 pods
+            if labels:
+                for label in labels:
+                    self.__create_pod_log(label=label)
+            else:
+                self.__create_pod_log(label=workload)
+            if database:
+                self.__create_pod_log(database=database)
+        workload_name = self.__environment_variables_dict.get('workload', '').replace('_', '-')
+        return os.path.join(self.__environment_variables_dict.get('run_artifacts_url', ''), f'{self.__get_run_artifacts_hierarchy(workload_name=workload_name)}-{self.__time_stamp_format}.tar.gz')
+
+    def __make_run_artifacts_tarfile(self, workload: str):
+        """
+        This method tar.gz log path and return the tar.gz path
+        :return:
+        """
+        run_artifacts_path = self.__environment_variables_dict.get('run_artifacts_path', '')
+        tar_run_artifacts_path = f"{run_artifacts_path}-{self.__time_stamp_format}.tar.gz"
+        with tarfile.open(tar_run_artifacts_path, mode='w:gz') as archive:
+            archive.add(run_artifacts_path, arcname=f'{workload}-{self.__time_stamp_format}', recursive=True)
+        return tar_run_artifacts_path
+
+    @logger_time_stamp
+    def upload_run_artifacts_to_s3(self, workload: str):
+        """
+        This method uploads log to s3
+        :param workload:
+        :return:
+        """
+        s3operations = S3Operations()
+        # change workload to key convention
+        workload = workload.replace('_', '-')
+        tar_run_artifacts_path = self.__make_run_artifacts_tarfile(workload)
+        run_artifacts_hierarchy = self.__get_run_artifacts_hierarchy()
+        upload_file = f"{workload}-{self.__time_stamp_format}.tar.gz"
+        s3operations.upload_file(file_name_path=tar_run_artifacts_path,
+                                 bucket=self.__environment_variables_dict.get('bucket', ''),
+                                 key=run_artifacts_hierarchy,
+                                 upload_file=upload_file)
+        # remove local run artifacts workload folder
+        path = self.__environment_variables_dict.get('run_artifacts_path', '')
+        # verify that its not empty path
+        if len(path) > 3 and path != '/' and path and tar_run_artifacts_path and os.path.isfile(tar_run_artifacts_path):
+            # remove run_artifacts_path
+            shutil.rmtree(path=self.__environment_variables_dict.get('run_artifacts_path', ''))
+            # remove tar.gz file
+            os.remove(path=tar_run_artifacts_path)
+
+
+
```

Comment:
This is at lesat mostly common code, not benchmark-operator code.  Don't refactor it right now, but this will be one more thing we need to refactor.